pkg/core/resources/model/resource.go

```diff
@@ -324,15 +334,21 @@ type ResourceMeta interface {
 // IsReferenced check if `refMeta` references with `refName` the entity `resourceMeta`
 // This is required because in multi-zone policies may have names different from the name they are defined with.
 func IsReferenced(refMeta ResourceMeta, refName string, resourceMeta ResourceMeta) bool {
+	if refMeta.GetMesh() != resourceMeta.GetMesh() {
+		return false
+	}
+
 	if len(refMeta.GetNameExtensions()) == 0 {
 		return equalNames(refMeta.GetMesh(), refName, resourceMeta.GetName())
 	}
 
-	if ns := refMeta.GetNameExtensions()[v1alpha1.KubeNamespaceTag]; ns != "" {
-		return equalNames(refMeta.GetMesh(), util_k8s.K8sNamespacedNameToCoreName(refName, ns), resourceMeta.GetName())
+	nsRef := refMeta.GetNameExtensions()[K8sNamespaceComponent]
+	nsRes := refMeta.GetNameExtensions()[K8sNamespaceComponent]
+	if nsRef == "" || nsRef != nsRes {
+		return false
 	}
 
-	return false
+	return equalNames(refMeta.GetMesh(), refName, resourceMeta.GetNameExtensions()[K8sNameComponent])
```

Comment:
So here we get these NameExtensions it's never been mentioned anywhere in the previous docs.

Reply:
It's just a name without nsSuffix, I'm avoiding name parsing (because dot is allowed in the name) so I took name from the name extension